README.md

```diff
@@ -1,2 +1,342 @@
+
+## Compatibility
+
+|                                   | Version               |
+|:---------------------------------:|:---------------------:|
+| Ballerina Language                | 2201.4.1              |
+| Java Development Kit (JDK)        | 11                    |
+
+## Example
+
+A simple EDI mapping is shown below (let's assume that this is saved in edi-mapping1.json file):
+
+````json
+{
+    "name": "SimpleOrder",
+    "delimiters" : {"segment" : "~", "field" : "*"},
+    "segments" : {
+        "HDR": {
+            "tag" : "header",
+            "fields" : [{"tag" : "orderId"}, {"tag" : "organization"}, {"tag" : "date"}]
+        },
+        "ITM": {
+            "tag" : "items",
+            "maxOccurances" : -1,
+            "fields" : [{"tag" : "item"}, {"tag" : "quantity", "dataType" : "int"}]
+        }
+    }
+}
+````
+
+Above mapping can be used to parse EDI documents with one HDR segment (mapped to "header") and any number of ITM segments (mapped to "items"). HDR segment contains three fields, which are mapped to "orderId", "organization" and "date". Each ITM segment contains two fields mapped to "item" and "quantity". Below is a sample EDI document that can be parsed using the above mapping (let's assume that below EDI is saved in edi-sample1.edi file):
+
+````edi
+HDR*ORDER_1201*ABC_Store*2008-01-01~
+ITM*A-250*12
+ITM*A-45*100
+ITM*D-10*58
+ITM*K-80*250
+ITM*T-46*28
+````
+
+### Code generation
+
+Ballerina records for the above the EDI mapping in edi-mapping1.json can be generated as follows (generated Ballerina records will be saved in orderRecords.bal):
+
+```
+java -jar edi.jar codegen edi-schema1.json orderRecords.bal
+```
+
+Generated Ballerina records for the above mapping are shown below:
+
+```ballerina
+type Header_Type record {|
+   string orderId?;
+   string organization?;
+   string date?;
+|};
+
+type Items_Type record {|
+   string item?;
+   int quantity?;
+|};
+
+type SimpleOrder record {|
+   Header_Type header;
+   Items_Type[] items?;
+|};
+```
+
+### Reading EDI files
+
+Below code reads the edi-sample1.edi into a json variable named "orderData" and then convert the orderData json to the generated record "SimpleOrder". Once EDI documents are mapped to the SimpleOrder record, any attribute in the EDI can be accessed using record's fields as shown in the example code below.
+
+````ballerina
+import ballerina/io;
+import balarinax/edi;
+
+public function main() returns error? {
+    EDIReader ediReader = check new(check io:fileReadJson("resources/edi-schema1.json"));
+    string ediText = check io:fileReadString("resources/edi-sample1.edi");
+    json orderData = check ediReader.readEDI(ediText);
+    io:println(orderData.toJsonString());
+
+    SimpleOrder order1 = check orderData.cloneWithType(SimpleOrder);
```

Comment:
Can't we make `readEDI` a dependently-typed function instead? Has to be an external function atm, but will allow writing

```ballerina
json orderJson = check ediReader.readEDI(ediText);

SimpleOrder orderRecord = check ediReader.readEDI(ediText);
```